### PR TITLE
[FEATURE] sitemap support for headless

### PR DESCRIPTION
**Pull request by twoldanski**

### CODE REVIEW

File: Classes/Seo/XmlSitemap/PagesXmlSitemapDataProvider.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace FriendsOfTYPO3\Headless\Seo\XmlSitemap;
+
+class PagesXmlSitemapDataProvider extends \TYPO3\CMS\Seo\XmlSitemap\PagesXmlSitemapDataProvider
+{
+    protected function defineUrl(array $data): array
+    {
+        $typoLinkConfig = [
+            'parameter' => $data['uid'],
+            'forceAbsoluteUrl' => 0,
```
Comment (lukaszuznanski) [inline Classes/Seo/XmlSitemap/PagesXmlSitemapDataProvider.php:22]: why we need this?

File: Classes/Seo/XmlSitemap/RecordsXmlSitemapDataProvider.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            $additionalParams,
+            '',
+            '&',
+            PHP_QUERY_RFC3986
+        );
+
+        $typoLinkConfig = [
+            'parameter' => $pageId,
+            'additionalParams' => $additionalParamsString ? '&' . $additionalParamsString : '',
+            'forceAbsoluteUrl' => 0,
```
Comment (lukaszuznanski) [inline Classes/Seo/XmlSitemap/RecordsXmlSitemapDataProvider.php:36]: also this

File: Classes/Seo/XmlSitemap/PagesXmlSitemapDataProvider.php (hunk 1 above)
Comment (twoldanski) [reply Classes/Seo/XmlSitemap/PagesXmlSitemapDataProvider.php:22]: we have to override forceAbsoluteUrl (default in CORE is 1), we have to replace API url with frontendBase url

File: Classes/Seo/XmlSitemap/RecordsXmlSitemapDataProvider.php (hunk 2 above)
Comment (twoldanski) [reply Classes/Seo/XmlSitemap/RecordsXmlSitemapDataProvider.php:36]: same as above

**Review by lukaszuznanski — APPROVED** [review]

**Pull request merged**